aws-iot-securityprofile/aws-iot-securityprofile.json

```diff
@@ -129,6 +131,28 @@
             "minimum": 0,
             "maximum": 65535
           }
+        },
+        "Number": {
+          "description": "The numeral value of a metric.",
+          "type": "number"
+        },
+        "Numbers": {
+          "description": "The numeral values of a metric.",
+          "type": "array",
+          "uniqueItems": true,
+          "insertionOrder": false,
+          "items": {
+            "type": "number"
+          }
+        },
+        "Strings": {
+          "description": "The string values of a metric.",
+          "type": "array",
+          "uniqueItems": true,
+          "insertionOrder": false,
+          "items": {
+            "type": "string"
```

Comment:
String values for Custom metric have a regex of their own, it might be good to quote the regex in CFN template just like we do for primary identifiers like SP name etc.

Reply:
Ass discussed, keeping the validations to minimum in cfn and leaving it to the service to validate.